File: py/src/practise150/p1_contains_duplicate_test.py
from src.practise150.p1_contains_duplicate import Solution


def test_contains_duplicate():
    test_cases = [([1, 2, 3, 1], True), ([1, 4, 3, 5], False)]
    s = Solution()
    for test_case in test_cases:
        result = s.contains_duplicate(test_case[0])

        assert result == test_case[1]

    for test_case in test_cases:
        result = s.contains_duplicate_2(test_case[0])

        assert result == test_case[1]

File: go/practise150/p1_contains_duplicate_test.go
package practise150

import "testing"

func Test_contains_duplicate(t *testing.T) {
	test_cases := []struct {
		input    []int
		expected bool
	}{
		{
			input:    []int{1, 2, 3, 1},
			expected: true,
		},
		{
			input:    []int{1, 4, 3, 5},
			expected: false,
		},
		{
			input:    []int{1, 2, 3, 11, 12, 4, 3, 2},
			expected: true,
		},
	}

	for _, test_case := range test_cases {
		t.Run("", func(t *testing.T) {
			actual := contains_duplicate(test_case.input)
			if actual != test_case.expected {
				t.Errorf("Checking for duplicates. Expected : %t, Actual : %t", test_case.expected, actual)
			}
		})
	}

}
